### ISSUE TITLE
Notes on Block decoding

### ISSUE THREAD

**Issue by decentralisedkev** [nspcc-dev/neo-go#85]
Some observations on the block decoding process.

After copying the leveldb file from neo-python and using the block.go file to deserialise the raw file, I was immediately met with a attribute usage encoding error.

When done with the chain.acc file, the error appeared around block 1842913.

I have crosschecked the neo-python leveldb file and the raw format returned from an rpc call and neo-python returns a different result for the same block. In particular, the first 4/8 bytes have to be cut, and the ending bytes do not match, maybe around 32.

**Comment by anthdm** [comment]
Nice found, will dive into it as we speak.

**Comment by anthdm** [comment]
@decentralisedkev If you could point me to the exact hash/index of the block that caused this problem I will love you forever.

**Comment by decentralisedkev** [comment]
@anthdm Let me check the PR, the solution should be in there

**Comment by decentralisedkev** [comment]
It's not there, I will DM you the script for the chain.acc file.

**Comment by anthdm** [comment]
@decentralisedkev I will start to port over your `wire` package to the master branch. I assume your work there will fix this issue.

**Comment by decentralisedkev** [comment]
It will. For those who are only on Github, this is being discussed on Slack.

**Comment by anthdm** [comment]
@kev I think the wire package would be a good start for this.

**Comment by hal0x2328** [comment]
@Kev Can we close this issue or does the problem remain?

**Comment by decentralisedkev** [comment]
@hal0x2328 The problem still remains on master, but not on v2dev. It's still uncertain as to what the plan is, in terms of the roadmap

**Comment by hal0x2328** [comment]
@decentralisedkev Yes, definitely would help to know the plan in terms of v2dev being merged into master. In the meantime, should we make a new issue to backport the fix into master?

**Comment by roman-khimov** [comment]
> When done with the chain.acc file, the error appeared around block 1842913

Not sure which network this number corresponds to, but I was able to get past 2M blocks synchronizing with testnet using current master (+ #396, to be fair). Will try mainnet.

**Comment by roman-khimov** [comment]
Now that I'm at more than 3M on mainnet:
```
INFO[75536] blockchain persist completed                  blockHeight=3036734 headerHeight=4286634 persisted=2 took=124.45137ms
```
I think it's safe to say that whatever the bug was here it no longer is relevant for the current master. Most probably, it was fixed in #320 or #328. Closing this one.
